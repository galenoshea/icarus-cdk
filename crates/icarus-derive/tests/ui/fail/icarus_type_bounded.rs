use icarus_derive::*;
use serde::{Deserialize, Serialize};
use candid::CandidType;

#[derive(IcarusType, Debug, Clone, Serialize, Deserialize, CandidType)]
#[icarus_storable(bounded, max_size = "1MB")]
struct BoundedData {
    content: String,
    timestamp: u64,
}

fn main() {}