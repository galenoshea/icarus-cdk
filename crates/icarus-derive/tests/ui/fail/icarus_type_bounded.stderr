error[E0046]: not all trait items implemented, missing: `into_bytes`
 --> tests/ui/fail/icarus_type_bounded.rs:5:10
  |
5 | #[derive(IcarusType, Debug, Clone, Serialize, Deserialize, CandidType)]
  |          ^^^^^^^^^^ missing `into_bytes` in implementation
  |
  = note: this error originates in the derive macro `IcarusType` (in Nightly builds, run with -Z macro-backtrace for more info)
